composites/AlgoliaSearch/SearchBar.js

```diff
@@ -20,28 +51,110 @@ const onSubmit = ( submitEvent, props ) => {
  * @returns {ReactElement} A div with the searchbar.
  * @constructor
  */
-const SearchBar = ( props ) => {
-	return (
-		<div className="wpseo-kb-search-search-bar">
-			<h2 id="wpseo-kb-search-heading">{ props.headingText }</h2>
-			<form onSubmit={ ( submitEvent ) => onSubmit( submitEvent, props ) }>
-				<input type="text" aria-labelledby="wpseo-kb-search-heading" defaultValue={ props.searchString }/>
-				<button type="submit" className="button wpseo-kb-search-search-button">{ props.searchButtonText }</button>
-			</form>
-		</div>
-	);
-};
+class SearchBar extends React.Component {
+
+	/**
+	 * Constructs the component and sets its initial state.
+	 *
+	 * @param {Object} props The props to use for this component.
+	 *
+	 * @constructor
+	 */
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			doRequest: false,
+			searchString: "",
+		};
+
+		this.doFormSubmission = debounce( ( searchString ) => {
+			this.props.submitAction( searchString );
+		}, 1000 );
+	}
+
+	/**
+	 * Cancel form submission on unmount.
+	 *
+	 * @returns {void}
+	 */
+	componentWillUnmount() {
+		this.doFormSubmission.cancel();
+	}
+
+	/**
+	 * Handles the change event on the SearchBar.
+	 *
+	 * @param {DOMEvent} event The event being triggered on the SearchBar.
+	 *
+	 * @returns {void}
+	 */
+	onSearchChange( event ) {
+		event.persist();
+		this.setState( { searchString: event.target.value }, () => {
+			this.doFormSubmission( this.state.searchString );
+		} );
+	}
+
+	/**
+	 * Handles the search bar form submit.
+	 *
+	 * @param {DOMEvent} event The event being triggered on the SearchBar.
+	 *
+	 * @returns {void}
+	 */
+	onSubmit( event ) {
+		event.preventDefault();
+		this.doFormSubmission.cancel();
+		this.props.submitAction( this.state.searchString );
+	}
+
+	/**
+	 * Renders the SearchBar component.
+	 *
+	 * @returns {ReactElement} The SearchBar component.
+	 */
+	render() {
+		const headingText = this.props.intl.formatMessage( messages.headingText );
+		return (
+			<SearchBarWrapper role="search">
+				<SearchHeading>
+					{ headingText }
+				</SearchHeading>
+				<form onSubmit={ this.onSubmit.bind( this ) }>
+					<SearchLabel htmlFor="search-input">
+						<Icon icon={ SearchIcon } color="inherit" size="30px" />
+						<span className="screen-reader-text">
+							{ headingText }
+						</span>
+					</SearchLabel>
+					<SearchBarInput
+						onChange={ this.onSearchChange.bind( this ) }
+						onSubmit={ ( event ) => {
```

Comment:
is this necessary? form submission is already prevented on the form itself, see `this.onSubmit()`